Various Enhancements for `testnet` Binary

jacderida opened the pull request:
- 48db17d **ci: bring benchmark chart workflow in line**

  The workflow for generating benchmark charts is very similar to the benchmark workflow that runs for
  PRs, but it was missed in the previous PR when the benchmark workflow was updated.

  This change brings the two in line with each other.

- 3d8732a **feat: remove network contacts from `testnet` bin**

  While we do still have the concept of contacting other peers--via the `--peer` argument or
  `SAFE_PEERS` environment variable--we are not using a 'network contacts' file any more, so we don't
  need this argument for the `testnet` binary.
  
- 9039d1a **refactor: obtain genesis peer id directly**

  Refactor the genesis launching process to use the gRPC generated code to obtain the genesis peer ID,
  rather than using the `safe_rpc_client` example. The latter was actually built during the process,
  which involved having the source code checked out. This is probably an assumption we shouldn't make
  of a released binary. It's intended to be able to work stand alone, without the source.

  BREAKING CHANGE: Also remove the `verify-nodes` feature in favour of just including this code in the
  binary. It only increases the size by a few megabytes and the binary is much more useful with this
  code.
  
- 71f8056 **tests: restore sn_testnet unit tests**

  The `sn_testnet` crate was more or less just copy/pasted from the previous safe_network repo and
  its behaviour was quickly changed for dealing with the new `safenode` binary. For quickness, the
  unit tests were subverted.

  These unit tests are now brought into line with the way the new `safenode` process operates and we
  can now keep them running for preventing regressions when things change.

- a4f4896 **feat: remove node directory management**

  The testnet binary is updated to remove any node directory management, which is now performed by the
  `safenode` process.

  This simplifies both the code and test cases, and also removes a few cases that are no longer
  necessary.
  
- 634c287 **feat: provide a `--clean` flag**

  Provides a `--clean` flag to remove all previous node root directories under the data directory,
  which is platform specific.
  
## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 06 Jul 23 20:14 UTC
This pull request includes the following changes:

1. In the `Cmd` struct, the `network_contacts_path` field has been removed. It was previously used with the 'join' command to specify the location of the network contacts file. The default location is now used if the value is not supplied.

2. In the `main` and `join_network` functions, the `network_contacts_path` parameter has been removed.

3. The `testnet.launch_nodes` function calls in both the `run_network` and `join_network` functions no longer include the `network_contacts_path` parameter.

4. The `check_testnet::run` function call within the `run_network` function has been removed.

5. The `init_tracing` function no longer returns a `Result` type.

This pull request also includes several changes in the `.github/workflows/generate-benchmark-charts.yml` file:

- Added `HEAPNODE_DATA_PATH` environment variable to set the data path for `heapnode`.
- Updated the names of job steps to accurately reflect their purposes.
- Updated the `run` command to download a 95MB file to be uploaded with the safe client.
- Updated the `run` command to start a heaptracked node instance with the appropriate data path.
- Updated the `run` command to start a heaptracked client instance with the appropriate data path.
- Updated the `run` command to upload the test data file.
- Updated the `run` command to find log files and create compressed archives.
- Updated the `run` command to analyze the heaptrack file for node memory usage.
- Updated the `name` and `value` of a JSON object to reflect node memory usage.
- Updated the `run` command to analyze the heaptrack file for client memory usage.
- Updated the `name` and `value` of JSON objects to reflect client memory usage.

In the `lib.rs` file, the following changes are present:

1. In the `TestnetBuilder` struct, the `build` method now returns a `Result<Testnet>` instead of `(Testnet, PathBuf)`. The network contacts path is no longer returned.

2. In the `TestnetBuilder` struct, the `build` method implementation has been updated to reflect the change in the method signature. The `network_contacts_path` variable has been removed from the implementation.

3. In the `Testnet` struct, the `launch_nodes` method no longer includes the `network_contacts_path` parameter in its signature or implementation.

4. In the `Testnet` struct, the `get_launch_args` method no longer includes the `_network_contacts_path` parameter in its implementation.

5. The tests for the `TestnetBuilder` and `Testnet` structs have been updated to match the changes in method signatures and implementations.

These changes simplify the code by removing unnecessary parameters and returning values.
<!-- reviewpad:summarize:end -->

lionel1704 reviewed (approved):
Looking good! :100:

jacderida committed:
feat: remove network contacts from `testnet` bin

While we do still have the concept of contacting other peers--via the `--peer` argument or
`SAFE_PEERS` environment variable--we are not using a 'network contacts' file any more, so we don't
need this argument for the `testnet` binary.

jacderida committed:
refactor: obtain genesis peer id directly

Refactor the genesis launching process to use the gRPC generated code to obtain the genesis peer ID,
rather than using the `safe_rpc_client` example. The latter was actually built during the process,
which involved having the source code checked out. This is probably an assumption we shouldn't make
of a released binary. It's intended to be able to work stand alone, without the source.

BREAKING CHANGE: Also remove the `verify-nodes` feature in favour of just including this code in the
binary. It only increases the size by a few megabytes and the binary is much more useful with this
code.

jacderida committed:
tests: restore sn_testnet unit tests

The `sn_testnet` crate was more or less just copy/pasted from the previous safe_network repo and
its behaviour was quickly changed for dealing with the new `safenode` binary. For quickness, the
unit tests were subverted.

These unit tests are now brought into line with the way the new `safenode` process operates and we
can now keep them running for preventing regressions when things change.

jacderida committed:
feat: remove node directory management

The testnet binary is updated to remove any node directory management, which is now performed by the
`safenode` process.

This simplifies both the code and test cases, and also removes a few cases that are no longer
necessary.

jacderida committed:
feat: provide a `--clean` flag

Provides a `--clean` flag to remove all previous node root directories under the data directory,
which is platform specific.

jacderida committed:
ci: bring benchmark chart workflow in line

The workflow for generating benchmark charts is very similar to the benchmark workflow that runs for
PRs, but it was missed in the previous PR when the benchmark workflow was updated.

This change brings the two in line with each other.
